libnavui-maneuver/src/test/java/com/mapbox/navigation/ui/maneuver/RoadShieldContentManagerTest.kt

```diff
@@ -0,0 +1,754 @@
+package com.mapbox.navigation.ui.maneuver
+
+import com.mapbox.bindgen.ExpectedFactory
+import com.mapbox.navigation.testing.MainCoroutineRule
+import com.mapbox.navigation.ui.maneuver.RoadShieldContentManager.Companion.CANCELED_MESSAGE
+import com.mapbox.navigation.ui.maneuver.model.Maneuver
+import com.mapbox.navigation.ui.maneuver.model.RoadShield
+import com.mapbox.navigation.ui.maneuver.model.RoadShieldComponentNode
+import com.mapbox.navigation.ui.maneuver.model.RoadShieldError
+import com.mapbox.navigation.ui.maneuver.model.RoadShieldResult
+import com.mapbox.navigation.ui.maneuver.model.TextComponentNode
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkObject
+import io.mockk.unmockkObject
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+/**
+ * The internal logic of [RoadShieldContentManager] requires us to have a very precise control over
+ * timing of coroutines so you'll see some paused dispatchers and granular timing advancement.
+ * You can read more about the technics used in
+ * [this guide](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-test/).
+ *
+ * Normally the `runBlockingTest` executes all coroutines eagerly, meaning that whenever a coroutine is launched, it's immediately executed.
+ * However, in a production environment that coroutine would most-likely be added back to the message queue which significantly impacts the logic.
+ * That's why this class uses the `pauseDispatcher` when needed to retain that type of the flow.
+ * Where appropriate, each test has 2 variants, one eager and one regular to make sure that we can handle both.
+ *
+ * Another important aspect is download delays introduce in some of the test to verify that we're not downloading things twice,
+ * that the queues are built correctly, and that cancellation works.
+ * To be able to control the elapsed time, we need to wrap invocations in another coroutine which is done in some of the tests.
+ */
+@ExperimentalCoroutinesApi
+class RoadShieldContentManagerTest {
```

Comment:
Tests need to be adjusted to the latest changes, they're currently failing https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-android/12611/workflows/56089cd0-0fb3-41ca-9d6f-a15236a14dbb/jobs/51546

Reply:
Fixed :+1: